flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/distance/EuclideanDistance.java

```diff
@@ -0,0 +1,259 @@
+package org.apache.flink.ml.classification.knn.distance;
+
+import org.apache.flink.ml.common.linalg.BLAS;
+import org.apache.flink.ml.common.linalg.DenseMatrix;
+import org.apache.flink.ml.common.linalg.DenseVector;
+import org.apache.flink.ml.common.linalg.MatVecOp;
+import org.apache.flink.ml.common.linalg.SparseVector;
+import org.apache.flink.ml.common.linalg.Vector;
+import org.apache.flink.util.Preconditions;
+
+import java.util.Arrays;
+
+/**
+ * Euclidean distance is the "ordinary" straight-line distance between two points in Euclidean
+ * space.
+ *
+ * <p>https://en.wikipedia.org/wiki/Euclidean_distance
+ *
+ * <p>Given two vectors a and b, Euclidean Distance = ||a - b||, where ||*|| means the L2 norm of
+ * the vector.
+ */
+public class EuclideanDistance extends BaseFastDistance {
```

Comment:
Can we re-use `EuclideanDistanceMeasure` for now? Otherwise, we will need to discuss the design for Flink ML infra classes such as `FastDistanceVectorData`, which will take sometime I guess.

Same for other Flink ML infra classes.

Reply:
Here, the "infra" is not Flink ML infra, it's only the infra of distance design. It's an inner infra of knn algorithm.
By this infra, algorithm developer can add new distance type, or add new accelerated method of distance calculation.

I think, this infra doesn't need discuss any more.

Reply:
after discuss with lindong, I agree this suggestion.
I have remove them already.

Reply:
Has the PR been updated with the latest change?

Reply:
yes